packages/blockfrost-client/test/Utxo/blockfrostUtxoProvider.test.ts

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable max-len */
+
+import { BlockFrostAPI, Responses } from '@blockfrost/blockfrost-js';
+import { Cardano, ProviderError, ProviderFailure } from '@cardano-sdk/core';
+import { blockfrostUtxoProvider } from '../../src/UtxoProvider/blockfrostUtxoProvider';
+jest.mock('@blockfrost/blockfrost-js');
+
+const generateUtxoResponseMock = (qty: number) =>
+  [...Array.from({ length: qty }).keys()].map((num) => ({
+    amount: [
+      {
+        quantity: String(50_928_877 + num),
+        unit: 'lovelace'
+      },
+      {
+        quantity: num + 1,
+        unit: 'b01fb3b8c3dd6b3705a5dc8bcd5a70759f70ad5d97a72005caeac3c652657675746f31333237'
+      }
+    ],
+    block: 'b1b23210b9de8f3edef233f21f7d6e1fb93fe124ba126ba924edec3043e75b46',
+    output_index: num,
+    tx_hash: '0f3abbc8fc19c2e61bab6059bf8a466e6e754833a08a62a6c56fe0e78f19d9d5',
+    tx_index: num
+  })) as Responses['address_utxo_content'];
+
+describe('blockfrostUtxoProvider', () => {
+  const apiKey = 'someapikey';
+
+  describe('healthCheck', () => {
+    it('returns ok if the service reports a healthy state', async () => {
+      BlockFrostAPI.prototype.health = jest.fn().mockResolvedValue({ is_healthy: true });
+      const blockfrost = new BlockFrostAPI({ network: 'preprod', projectId: apiKey });
+      const provider = blockfrostUtxoProvider(blockfrost);
+      expect(await provider.healthCheck()).toEqual({ ok: true });
+    });
+    it('returns not ok if the service reports an unhealthy state', async () => {
+      BlockFrostAPI.prototype.health = jest.fn().mockResolvedValue({ is_healthy: false });
+      const blockfrost = new BlockFrostAPI({ network: 'preprod', projectId: apiKey });
+      const provider = blockfrostUtxoProvider(blockfrost);
+      expect(await provider.healthCheck()).toEqual({ ok: false });
+    });
+    it('throws a typed error if caught during the service interaction', async () => {
+      BlockFrostAPI.prototype.health = jest
+        .fn()
+        .mockRejectedValue(new ProviderError(ProviderFailure.Unknown, new Error('Some error')));
+      const blockfrost = new BlockFrostAPI({ network: 'preprod', projectId: apiKey });
+      const provider = blockfrostUtxoProvider(blockfrost);
+      await expect(provider.healthCheck()).rejects.toThrowError(ProviderError);
+    });
+  });
+
+  describe('utxoByAddresses', () => {
```

Comment:
Please add a couple tests for mapping Blockfrost/connection errors into `ProviderError`
```suggestion
  describe('utxoByAddresses', () => {
     it('throws ProviderError when...');
```

Reply:
@mkazlauskas To be honest, my scope for this task was to revert the existing commits and make it work.
If this merge is not meeting our standards, I would wait and merge it with the scope of https://input-output.atlassian.net/browse/LW-11239

Reply:
Please capture any remaining work/change requests in this ticket then